greeclimate/device.py

```diff
@@ -344,10 +348,16 @@ def _convert_to_units(self, value, bit):
         if self.temperature_units != TemperatureUnits.F.value:
             return value
 
-        if value < TEMP_MIN or value > TEMP_MAX:
+        if value < TEMP_MIN_TABLE or value > TEMP_MAX_TABLE:
             raise ValueError(f"Specified temperature {value} is out of range.")
 
-        f = next(t for t in TEMP_TABLE if t["temSet"] == value and t["temRec"] == bit)
+        matching_temSet = [t for t in TEMP_TABLE if t["temSet"] == value]
+
+        try:
+            f = next(t for t in matching_temSet if t["temRec"] == bit)
+        except StopIteration:
+            f = matching_temSet[0]
```

Comment:
Please add test cases covering these lines.

Reply:
I haven't gotten around to that, I'll take a look at this shortly.

Reply:
Also, do you have any clue why some units have discrepancies between the temSets and temRecs? Since this solution has worked fine for me on HomeAssistant, it isn't very important to actually implement, but addressing this problem would fix the root of this hacky solution.

Reply:
No, not sure. This whole approach to supporting F feels like Gree added a feature that while trying not to break C support on existing hardware. It's seems likely that the conversion is not computed on need, I could never find a functional formula that just worked and so had to pre-compute the table for the lookup to work at all.